creator-node/src/snapbackSM/peerSetManager.js

```diff
@@ -0,0 +1,221 @@
+const axios = require('axios')
+
+const { logger } = require('../logging')
+
+// TODO: config var
+const PEER_HEALTH_CHECK_REQUEST_TIMEOUT = 2000
+
+class PeerSetManager {
+  constructor ({ discoveryProviderEndpoint, creatorNodeEndpoint }) {
+    this.discoveryProviderEndpoint = discoveryProviderEndpoint
+    this.creatorNodeEndpoint = creatorNodeEndpoint
+  }
+
+  log (msg) {
+    logger.info(`SnapbackSM:::PeerSetManager: ${msg}`)
+  }
+
+  logError (msg) {
+    logger.error(`SnapbackSM:::PeerSetManager ERROR: ${msg}`)
+  }
+
+  /**
+   * Retrieves node users, paritions it into a slice, computes the peer set of the node users,
+   * and performs a health check on the peer set.
+   * @param {Object[]} nodeUsers array of objects of schema { primary, secondary1, secondary2, user_id, wallet }
+   * @returns the unhealthy peers in a Set
+   *
+   * @note consider returning healthy set?
+   */
+  async getUnhealthyPeers (nodeUsers) {
+    // Compute content node peerset from nodeUsers (all nodes that are in a shared replica set with this node)
+    let peerSet = this.computeContentNodePeerSet(nodeUsers)
+
+    /**
+     * Determine health for every peer & build list of unhealthy peers
+     * TODO: change from sequential to chunked parallel
+     */
+    const unhealthyPeers = new Set()
+
+    for await (const peer of peerSet) {
+      try {
+        // TODO: consider using the response
+        await this.determinePeerHealth(peer)
```

Comment:
this call is try/catched in case the axios req does not respond with a 200

Reply:
we should add this as an explicit code comment

Reply:
I don't think this is necessary

Reply:
there is already a comment in `determinePeerHealth()` that says if the axios req fails, it throws 
https://github.com/AudiusProject/audius-protocol/pull/1502/files/0ab1cea70368be1e941dcb2efcf191d3788d4950#diff-5ea38207482dfa410830e31be105407f8c6c8fc22d59036ce3647791e28ed24cR211

Reply:
yeah i saw that, i think it would improve the readability of this function to add that here as well but its nbd